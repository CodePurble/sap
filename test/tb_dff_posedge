#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x55856c39f330 .scope module, "tb_dff_posedge" "tb_dff_posedge" 2 1;
 .timescale 0 0;
v0x55856c3c6980_0 .var "clear", 0 0;
v0x55856c3c6a40_0 .var "clk", 0 0;
v0x55856c3c6ae0_0 .var "d", 0 0;
v0x55856c3c6bb0_0 .var "preset", 0 0;
v0x55856c3c6c80_0 .net "q", 0 0, v0x55856c3c66f0_0;  1 drivers
v0x55856c3c6d70_0 .net "qbar", 0 0, L_0x55856c3c6e40;  1 drivers
S_0x55856c39f4c0 .scope module, "uut" "dff_posedge" 2 5, 3 2 0, S_0x55856c39f330;
 .timescale 0 0;
    .port_info 0 /OUTPUT 1 "q";
    .port_info 1 /OUTPUT 1 "qbar";
    .port_info 2 /INPUT 1 "d";
    .port_info 3 /INPUT 1 "preset";
    .port_info 4 /INPUT 1 "clear";
    .port_info 5 /INPUT 1 "clk";
L_0x55856c3c6e40 .functor NOT 1, v0x55856c3c66f0_0, C4<0>, C4<0>, C4<0>;
v0x55856c377d50_0 .net "clear", 0 0, v0x55856c3c6980_0;  1 drivers
v0x55856c3c64d0_0 .net "clk", 0 0, v0x55856c3c6a40_0;  1 drivers
v0x55856c3c6590_0 .net "d", 0 0, v0x55856c3c6ae0_0;  1 drivers
v0x55856c3c6630_0 .net "preset", 0 0, v0x55856c3c6bb0_0;  1 drivers
v0x55856c3c66f0_0 .var "q", 0 0;
v0x55856c3c6800_0 .net "qbar", 0 0, L_0x55856c3c6e40;  alias, 1 drivers
E_0x55856c3a1200 .event posedge, v0x55856c3c64d0_0, v0x55856c377d50_0, v0x55856c3c6630_0;
    .scope S_0x55856c39f4c0;
T_0 ;
    %wait E_0x55856c3a1200;
    %load/vec4 v0x55856c3c6630_0;
    %pad/u 32;
    %cmpi/e 1, 0, 32;
    %jmp/0xz  T_0.0, 4;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0x55856c3c66f0_0, 0;
    %jmp T_0.1;
T_0.0 ;
    %load/vec4 v0x55856c377d50_0;
    %pad/u 32;
    %cmpi/e 1, 0, 32;
    %jmp/0xz  T_0.2, 4;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0x55856c3c66f0_0, 0;
    %jmp T_0.3;
T_0.2 ;
    %load/vec4 v0x55856c3c6590_0;
    %assign/vec4 v0x55856c3c66f0_0, 0;
T_0.3 ;
T_0.1 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0x55856c39f330;
T_1 ;
    %vpi_call 2 9 "$dumpfile", "../simulation/tb_dff_posedge.vcd" {0 0 0};
    %vpi_call 2 10 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x55856c39f330 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55856c3c6bb0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55856c3c6980_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55856c3c6ae0_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55856c3c6ae0_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55856c3c6ae0_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x55856c3c6ae0_0, 0, 1;
    %delay 10, 0;
    %vpi_call 2 24 "$finish" {0 0 0};
    %end;
    .thread T_1;
    .scope S_0x55856c39f330;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x55856c3c6a40_0, 0, 1;
    %pushi/vec4 10, 0, 32;
T_2.0 %dup/vec4;
    %pushi/vec4 0, 0, 32;
    %cmp/s;
    %jmp/1xz T_2.1, 5;
    %jmp/1 T_2.1, 4;
    %pushi/vec4 1, 0, 32;
    %sub;
    %load/vec4 v0x55856c3c6a40_0;
    %inv;
    %store/vec4 v0x55856c3c6a40_0, 0, 1;
    %delay 10, 0;
    %jmp T_2.0;
T_2.1 ;
    %pop/vec4 1;
    %end;
    .thread T_2;
    .scope S_0x55856c39f330;
T_3 ;
    %vpi_call 2 39 "$monitor", "d = %b, clk = %b, q = %b", v0x55856c3c6ae0_0, v0x55856c3c6a40_0, v0x55856c3c6c80_0 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "/home/ramprakash/repos/sap/test/tb_dff_posedge.v";
    "../hdl/dff_posedge.v";
