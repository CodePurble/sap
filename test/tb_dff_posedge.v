module tb_dff_posedge;
    reg d, clk, preset, clear;
    wire q, qbar;

    dff_posedge uut(.q(q), .qbar(qbar), .d(d), .preset(preset), .clear(clear), .clk(clk));

    initial
    begin
        $dumpfile("./simulation/tb_dff_posedge.vcd");
        $dumpvars(0, tb_dff_posedge);
        preset = 1'b1;
        clear = 1'b1;
        d = 1'b1;
        #10;

        clear = 1'b0;
        d = 1'b0;
        #10;

        d = 1'b0;
        #10;

        d = 1'b1;
        #10;
        $finish;
    end

    initial
    begin
        clk = 1'b0;
        repeat(10)
        begin
            clk = ~clk;
            #10;
        end
    end

    initial
    begin
        $monitor("d = %b, clk = %b, q = %b", d, clk, q);
    end
endmodule
