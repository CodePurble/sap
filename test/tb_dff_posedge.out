#! /usr/bin/vvp
:ivl_version "11.0 (stable)" "(v11_0)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/ivl/system.vpi";
:vpi_module "/usr/lib/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/ivl/va_math.vpi";
S_0x5587107c3460 .scope module, "tb_dff_posedge" "tb_dff_posedge" 2 1;
 .timescale 0 0;
v0x5587107eaa30_0 .var "clear", 0 0;
v0x5587107eaaf0_0 .var "clk", 0 0;
v0x5587107eab90_0 .var "d", 0 0;
v0x5587107eac60_0 .var "preset", 0 0;
v0x5587107ead30_0 .net "q", 0 0, v0x5587107ea7a0_0;  1 drivers
v0x5587107eae20_0 .net "qbar", 0 0, L_0x5587107eaef0;  1 drivers
S_0x55871079bcf0 .scope module, "uut" "dff_posedge" 2 5, 3 2 0, S_0x5587107c3460;
 .timescale 0 0;
    .port_info 0 /OUTPUT 1 "q";
    .port_info 1 /OUTPUT 1 "qbar";
    .port_info 2 /INPUT 1 "d";
    .port_info 3 /INPUT 1 "preset";
    .port_info 4 /INPUT 1 "clear";
    .port_info 5 /INPUT 1 "clk";
L_0x5587107eaef0 .functor NOT 1, v0x5587107ea7a0_0, C4<0>, C4<0>, C4<0>;
v0x55871079bf20_0 .net "clear", 0 0, v0x5587107eaa30_0;  1 drivers
v0x5587107ea580_0 .net "clk", 0 0, v0x5587107eaaf0_0;  1 drivers
v0x5587107ea640_0 .net "d", 0 0, v0x5587107eab90_0;  1 drivers
v0x5587107ea6e0_0 .net "preset", 0 0, v0x5587107eac60_0;  1 drivers
v0x5587107ea7a0_0 .var "q", 0 0;
v0x5587107ea8b0_0 .net "qbar", 0 0, L_0x5587107eaef0;  alias, 1 drivers
E_0x5587107c5200 .event posedge, v0x5587107ea580_0, v0x55871079bf20_0, v0x5587107ea6e0_0;
    .scope S_0x55871079bcf0;
T_0 ;
    %wait E_0x5587107c5200;
    %load/vec4 v0x5587107ea6e0_0;
    %pad/u 32;
    %cmpi/e 1, 0, 32;
    %jmp/0xz  T_0.0, 4;
    %pushi/vec4 1, 0, 1;
    %assign/vec4 v0x5587107ea7a0_0, 0;
    %jmp T_0.1;
T_0.0 ;
    %load/vec4 v0x55871079bf20_0;
    %pad/u 32;
    %cmpi/e 1, 0, 32;
    %jmp/0xz  T_0.2, 4;
    %pushi/vec4 0, 0, 1;
    %assign/vec4 v0x5587107ea7a0_0, 0;
    %jmp T_0.3;
T_0.2 ;
    %load/vec4 v0x5587107ea640_0;
    %assign/vec4 v0x5587107ea7a0_0, 0;
T_0.3 ;
T_0.1 ;
    %jmp T_0;
    .thread T_0;
    .scope S_0x5587107c3460;
T_1 ;
    %vpi_call 2 9 "$dumpfile", "../simulation/tb_dff_posedge.vcd" {0 0 0};
    %vpi_call 2 10 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x5587107c3460 {0 0 0};
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5587107eac60_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5587107eaa30_0, 0, 1;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x5587107eab90_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5587107eab90_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5587107eab90_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0x5587107eab90_0, 0, 1;
    %delay 10, 0;
    %vpi_call 2 24 "$finish" {0 0 0};
    %end;
    .thread T_1;
    .scope S_0x5587107c3460;
T_2 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5587107eaaf0_0, 0, 1;
    %pushi/vec4 10, 0, 32;
T_2.0 %dup/vec4;
    %pushi/vec4 0, 0, 32;
    %cmp/s;
    %jmp/1xz T_2.1, 5;
    %jmp/1 T_2.1, 4;
    %pushi/vec4 1, 0, 32;
    %sub;
    %load/vec4 v0x5587107eaaf0_0;
    %inv;
    %store/vec4 v0x5587107eaaf0_0, 0, 1;
    %delay 10, 0;
    %jmp T_2.0;
T_2.1 ;
    %pop/vec4 1;
    %end;
    .thread T_2;
    .scope S_0x5587107c3460;
T_3 ;
    %vpi_call 2 39 "$monitor", "d = %b, clk = %b, q = %b", v0x5587107eab90_0, v0x5587107eaaf0_0, v0x5587107ead30_0 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tb_dff_posedge.v";
    "../hdl/dff_posedge.v";
